apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/match/HierarchyMatch.java

```diff
@@ -46,6 +47,11 @@ private HierarchyMatch(String[] parentTypes) {
         this.parentTypes = parentTypes;
     }
 
+    private HierarchyMatch(String[] parentTypes, boolean interfaceOnlyMatch) {
+        this(parentTypes);
+        this.interfaceOnlyMatch = interfaceOnlyMatch;
+    }
+
```

Comment:
Usually, we don't recommend to change the core API.

Reply:
RabbitMQ client pretty much leaves the message handling implementation to a custom class or a framework like Camel's RabbitMQ module or a Spring. We need to discover and instrument implementation classes of com.rabbitmq.client.Consumer

We cannot use HierarchyMtach as is. if we did, in this case, it might discover an interface that extends from com.rabbitmq.client.Consumer which will result in undesired behavior

So, HierarchyMtach is lacking and ideally should have a feature to just match interfaces only.

Reply:
If an interface extends, then what? You can't intercept an interface, because nothing invoked there. Wha is your point? And what is instrumentation for interface only meaning? How to enhance an interface?

Reply:
Follow this...

If I give an interface as input (e.g. com.rabbitmq.client.Consumer), I want classes that are implementing this interface (com.rabbitmq.client.Consumer) to be instrumented. I do not want interfaces extending from com.rabbitmq.client.Consumer instrumented. Is it possible with existing core API?

Reply:
Yes, that is `DeclaredInstanceMethodsInterceptPoint` for.

Reply:
@viswaramamoorthy Have you tried this?

Reply:
I am not how to make it work with DeclaredInstanceMethodsInterceptPoint. I do not know much about the code. 

I have removed my changes in HierarchyMatch class (which you are saying, I should not). I am directly using HierarchyMatch as is and instrumenting RabbitMQ interface and its underlying implementation. It is working,.

Review this. if it is fine, I can do changes recommended for legacy test case migration.

If this is not fine, you can make changes to this PR to fix the issue I have reported. Let me know if you have questions on the issue that I reported.

Reply:
Replace `InstanceMethodsInterceptPoint` to `DeclaredInstanceMethodsInterceptPoint` in your instrumentation define. It is an easy change. They you could test again. Then in the hierarchy case, it should workw as your exepected. @viswaramamoorthy

This interceptor point mean this method must be implemented in this method, rather that existing the parent class.

I think you are closing the right one.